### ISSUE TITLE
Detect redundant unions with buffer/bytes types?

### ISSUE THREAD

**Issue by Akuli** [PyCQA/flake8-pyi#414]
Currently this annotation doesn't cause a lint error: (taken from https://github.com/python/typeshed/pull/10432 )

```python
from _typeshed import ReadOnlyBuffer

def farm_msg(__farm_name: str, __message: str | bytes | ReadOnlyBuffer) -> None: ...
```

If I understand the buffer situation correctly, `bytes | ReadOnlyBuffer` is equivalent to `ReadOnlyBuffer`, because typeshed claims that `bytes` implements `__buffer__` on all Python versions.

It feels like a good idea to detect unions where bytes types are used like this, similar to `Literal[1] | int` and other checks we already have. But it feels like a bad idea to make flake8-pyi aware of `_typeshed.ReadOnlyBuffer` etc and their subtyping rules. I'm not sure.

**Comment by AlexWaygood** [comment]
> It feels like a good idea to detect unions where bytes types are used like this, similar to `Literal[1] | int` and other checks we already have. But it feels like a bad idea to make flake8-pyi aware of `_typeshed.ReadOnlyBuffer` etc and their subtyping rules. I'm not sure.

A general-purpose way of detecting redundant unions could only ever be implemented by a type checker, since they're much better equipped to resolve imports across modules than we'll ever be. We have a much simpler design that only analyses a module at a time, and I'd like to keep it that way (I'm not sure flake8's plugin system even would allow us to analyse multiple modules at a time). So we'll need to weigh up whether it's worth special-casing these specific types from `_typeshed` or not.

**Comment by Avasam** [comment]
Relevant pyright discussion: https://github.com/microsoft/pyright/discussions/3949

**Comment by AlexWaygood** [comment]
> So we'll need to weigh up whether it's worth special-casing these specific types from `_typeshed` or not.

My feeling is that this _probably_ doesn't come up enough for it to be worth adding this special-casing — WDYT?
